### PR TITLE
CI improvements

### PR DESCRIPTION
**Pull request by bamarni**
- bump python to 3.6
- Run Travis CI builds on MacOS too
- Add flake8 / pylint and fix code accordingly
- Add 'make lint' target

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,22 @@
...
-    - 3.6
-  install:
-    - make
-  script:
-    - make test
+language: python
+
+matrix:
+  include:
+    - os: linux
```
Comment (adamtheturtle) [inline .travis.yml:5]: Keep in mind that the current CLI supports Windows which is not supported by Travis CI.

File: .travis.yml [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+    - os: osx
+      # workaround for https://github.com/travis-ci/travis-ci/issues/2312
+      language: generic
+      before_install:
```
Comment (adamtheturtle) [inline .travis.yml:9]: Does this mean that macOS builds may have a different version of Python to Linux builds, and importantly, different to what the documentation says is required?

File: Makefile [hunk 3]
```diff
@@ -2,6 +2,11 @@ deps:
 	pip install -e .[dev]
 .PHONY: deps
 
-test:
+lint:
+	flake8 --verbose dcos_cli/ tests/
+	pylint --disable=missing-docstring,cyclic-import dcos_cli/ tests/
```
Comment (adamtheturtle) [inline Makefile]: `pylint` configuration inevitably becomes long. I recommend using a pylint configuration file, but that can wait 😄

File: dcos_cli/cli/cluster/__init__.py [hunk 4]
```diff
@@ -1,14 +1,14 @@
...
     """Manage your DC/OS clusters."""
     pass
 
-import dcos_cli.cli.cluster.attach # noqa: E402
-import dcos_cli.cli.cluster.list # noqa: E402
-import dcos_cli.cli.cluster.remove # noqa: E402
-import dcos_cli.cli.cluster.rename # noqa: E402
-import dcos_cli.cli.cluster.setup # noqa: E402
+# pylint: disable=wrong-import-position
+import dcos_cli.cli.cluster.attach # noqa
```
Comment (adamtheturtle) [inline dcos_cli/cli/cluster/__init__.py]: Why remove the specifics of the `noqa`? Using specific codes allows `flake8` to report issues that you didn't mean to ignore.

I don't understand why these are imported late. Please add a comment if there is a good reason, or move them to the top of the file.

File: .travis.yml (hunk 1 above)
Comment (bamarni) [reply .travis.yml:5]: That's right, I will take a look at https://www.appveyor.com in the next steps.

File: .travis.yml (hunk 2 above)
Comment (bamarni) [reply .travis.yml:9]: Good point, it seems to be 3.6 from the build console logs. I'll look for a way to stick to 3.5 there or to bump requirements (https://github.com/bamarni/dcos_cli/issues/2).

File: dcos_cli/cli/cluster/__init__.py (hunk 4 above)
Comment (bamarni) [reply dcos_cli/cli/cluster/__init__.py]: I was missing the noqa F401. As for these imports it's the same reason than here (https://github.com/mesosphere/dcos-security-cli/blob/master/dscli/__init__.py#L99).

I will check to see if there is another way to do this, that'd also allow to get rid of linter exceptions for them.

Comment (bamarni) [reply dcos_cli/cli/cluster/__init__.py]: I've switched to another way, cf. last commit.

Comment (adamtheturtle) [reply dcos_cli/cli/cluster/__init__.py]: 🙏

**Comment by bamarni** [comment]
@adamtheturtle : I'll merge this on once the build is green, as I think I've addressed all your comments, thanks for the review.

**Pull request merged**